integration-test.py

```diff
@@ -311,6 +311,25 @@ def assert_at_least_n_features(z, x, y, layer, properties, n):
                 "%r, found only %d" % (n, properties, num_matching)
 
 
+def assert_less_than_n_features(z, x, y, layer, properties, n):
+    """
+    Downloads a tile and checks that it contains less than `n` features which
+    match the given `properties`.
+    """
+    with features_in_tile_layer(z, x, y, layer) as features:
+        num_features, num_matching = count_matching(
+            features, properties)
+
+        if num_features >= n:
```

Comment:
I'm not sure that this fits with the description of the function.

For example, if a tile has 10 features, but only one matches the `properties`, and we test `assert_less_than_n_features(z, x, y, layer, properties, 2)`, then I would expect the test to pass. But the test on the total number of features here means that it won't.